scripts/tt-console/console.c

```diff
@@ -664,7 +671,17 @@ static int loop(struct console *const cons)
 		goto out;
 	}
 
-	I("Press Ctrl-a,x to quit");
+	/*
+	 * Check if STDOUT is a tty. If not, we should simply stream binary
+	 * data the file without any processing.
+	 */
+	if (isatty(STDOUT_FILENO)) {
+		using_tty = true;
+		I("Press Ctrl-a,x to quit");
+	} else {
+		fprintf(stderr, "Not a tty, streaming binary data to STDOUT. "
+				"Press Ctrl-C to quit.\n");
+	}
```

Comment:
I would undo this bit - you can definitely stream binary data even if it is an interactive console.

Additionally, console interactivity is usually determined by whether or not the input is a tty.